docker-entrypoint.sh

```diff
@@ -24,6 +24,10 @@ if ! which -- "${1}"; then
 	# Wait exits immediately on signals which have traps set. Store return value and wait
 	# again for all jobs to actually complete before continuing.
 	wait $! || RETVAL=$?
+	while [ ${RETVAL} = 129 ] ; do
+	  # Refressh signal HUP received. Continue waiting for signals.
+	  wait $! || RETVAL=$?
+	done
```

Comment:
@zstadler what does this actually do? why after HUP is received are we waiting for more signals?

Reply:
but it doesn't actually reload the service, it just starts waiting again. I was thinking making a PR to replace it with something like this https://github.com/maptiler/tileserver-gl/issues/629#issuecomment-1292352424

Reply:
```
trap refresh HUP
```
causes `refresh` to be executed when the shell process receives the HUP signal.

```
refresh() {
	SIGNAL=$(( $? - 128 ))
	echo "Caught signal ${SIGNAL}, refreshing"
	kill -s ${SIGNAL} $(pidof node) 2>/dev/null
}
```
causes the HUP signal to be sent to the `tileserver-gl` process.

```
wait $! || RETVAL=$?
```
keeps the shell process alive in order to relay any `HUP` signals to it.

```
while [ ${RETVAL} = 129 ] ; do
  # Refressh signal HUP received. Continue waiting for signals.
  wait $! || RETVAL=$?
done
```
The `wait` command exits once a signal is received. When a HUP signal is received, `RETVAL` is 129 and `wait` is executed again and the shell process waits for the next signal to come.